go/integTest/shared_commands_test.go

```diff
@@ -161,6 +161,67 @@ func (suite *GlideTestSuite) TestSetWithOptions_UpdateExistingExpiry() {
 	})
 }
 
+func (suite *GlideTestSuite) TestGetEx_existingAndNonExistingKeys() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := "TestGetEx_ExisitingKey"
+		suite.verifyOK(client.Set(key, initialValue))
+
+		result, err := client.GetEx(key)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), initialValue, result.Value())
+
+		key = "TestGetEx_NonExisitingKey"
+		result, err = client.Get(key)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), "", result.Value())
+	})
+}
+
+func (suite *GlideTestSuite) TestGetExWithOptions_PersistKey() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := "TestGetExWithOptions_PersistKey"
+		suite.verifyOK(client.Set(key, initialValue))
+
+		opts := &api.GetExOptions{Expiry: &api.Expiry{Type: api.Milliseconds, Count: uint64(2000)}}
+		result, err := client.GetExWithOptions(key, opts)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), initialValue, result.Value())
+
+		result, err = client.Get(key)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), initialValue, result.Value())
+
+		time.Sleep(1000 * time.Millisecond)
+
+		opts = &api.GetExOptions{Expiry: &api.Expiry{Type: api.Persist}}
+		result, err = client.GetExWithOptions(key, opts)
+		assert.Nil(suite.T(), err)
+		assert.Equal(suite.T(), initialValue, result.Value())
+	})
+}
+
+func (suite *GlideTestSuite) TestGetExWithOptions_UpdateExpiry() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
+		key := "TestGetExWithOptions_UpdateExpiry"
+		suite.verifyOK(client.Set(key, initialValue))
+
+		opts := &api.GetExOptions{Expiry: &api.Expiry{Type: api.Milliseconds, Count: uint64(2000)}}
```

Comment:
Maybe its the standard in `go`, but it seems a bit cumbersome. Can we provide convenient API for creating `GetExOptions` ?

Is it common to use Builder pattern in `go`? I did saw blogs about Builder in `go` so it seems acceptable, and IMO, it looks better:

```go
NewGetOptionsBuilder().SetExpiryType(api.Milliseconds).SetCount(2000)
```

WDYT?

Reply:
Modified.